econml/tests/test_automated_ml.py

```diff
@@ -0,0 +1,235 @@
+from econml.automated_ml import addAutomatedML
+from econml.tests.test_statsmodels import _summarize
+import econml.tests.utilities  # bugfix for assertWarns
+import copy
+import logging
+from econml.data.dgps import ihdp_surface_B
+from azureml.train.automl.exceptions import ClientException
+from azureml.core.authentication import AzureCliAuthentication
+import os
+
+AutomatedTLearner = addAutomatedML(TLearner)
+AutomatedSLearner = addAutomatedML(SLearner)
+AutomatedXLearner = addAutomatedML(XLearner)
+AutomatedDomainAdaptationLearner = \
+    addAutomatedML(DomainAdaptationLearner)
+AutomatedDRLearner = addAutomatedML(DRLearner)
+AutomatedDMLCateEstimator = addAutomatedML(DMLCateEstimator)
+AutomatedLinearDMLCateEstimator = addAutomatedML(LinearDMLCateEstimator)
+AutomatedSparseLinearDMLCateEstimator = \
+    addAutomatedML(SparseLinearDMLCateEstimator)
+AutomatedKernelDMLCateEstimator = addAutomatedML(KernelDMLCateEstimator)
+AutomatedNonParamDMLCateEstimator = \
+    addAutomatedML(NonParamDMLCateEstimator)
+AutomatedForestDMLCateEstimator = addAutomatedML(ForestDMLCateEstimator)
+
+# all solutions to underdetermined (or exactly determined) Ax=b are given by A⁺b+(I-A⁺A)w for some arbitrary w
+# note that if Ax=b is overdetermined, this will raise an assertion error
+
+
+def rand_sol(A, b):
+    """Generate a random solution to the equation Ax=b."""
+    assert np.linalg.matrix_rank(A) <= len(b)
+    A_plus = np.linalg.pinv(A)
+    x = A_plus @ b
+    return x + (np.eye(x.shape[0]) - A_plus @ A) @ np.random.normal(size=x.shape)
+
+
+AUTOML_SETTINGS_REG = {
+    'experiment_timeout_minutes': 1,
+    'enable_early_stopping': True,
+    'iteration_timeout_minutes': 1,
+    'max_cores_per_iteration': 1,
+    'n_cross_validations': 2,
+    'preprocess': False,
+    'featurization': 'off',
+    'enable_stack_ensemble': False,
+    'enable_voting_ensemble': False,
+    'primary_metric': 'normalized_mean_absolute_error',
+}
+
+AUTOML_SETTINGS_CLF = {
+    'experiment_timeout_minutes': 1,
+    'enable_early_stopping': True,
+    'iteration_timeout_minutes': 1,
+    'max_cores_per_iteration': 1,
+    'n_cross_validations': 2,
+    'enable_stack_ensemble': False,
+    'preprocess': False,
+    'enable_voting_ensemble': False,
+    'featurization': 'off',
+    'primary_metric': 'AUC_weighted',
+}
+
+
+AUTOML_CONFIG_REG = EconAutoMLConfig(task='regression',
+                                     debug_log='automl_errors.log',
+                                     enable_onnx_compatible_models=True, model_explainability=True,
+                                     **AUTOML_SETTINGS_REG)
+
+AUTOML_CONFIG_CLF = EconAutoMLConfig(task='classification',
+                                     debug_log='automl_errors.log',
+                                     enable_onnx_compatible_models=True, model_explainability=True,
+                                     **AUTOML_SETTINGS_CLF)
+
+AUTOML_CONFIG_LINEAR_REG = EconAutoMLConfig(task='regression',
```

Comment:
There appears to be no difference between the LINEAR and SAMPLE_WEIGHT variants defied here...